content/blog/azure-pipelines-scaler.md

```diff
@@ -0,0 +1,224 @@
+azdevops-deployment-5854bbbf84-r86qv   1/1     Running   0          12m
+azdevops-deployment-5854bbbf84-tm47k   1/1     Running   0          36s
+```
+
+![deployment agents](/img/blog/azure-pipelines-scaler/deployment-agents-autoscaled.png)
+
+## Run the self-hosted agent as a Job
+
+When running your agents as a deployment you have no control on which pod gets killed when scaling down. ([see KEDA docs](https://keda.sh/docs/1.4/concepts/scaling-deployments/#long-running-executions))
+
+If you run your agents as a `Job`, KEDA will start a Kubernetes job for each job that is in the agent pool queue. The agents will accept one job when they are started and terminate afterwards.
+You also achieve fully isolated build environments when using jobs, since an agent is always created for each job.
+
+The following manifest is an example of a `ScaledJob` combined with the Azure Pipelines agent.
+You have to use a modified image for this where the agent terminates itself after running a job. ([docs](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/v2-linux#run-once))
+
+```yaml
+apiVersion: keda.sh/v1alpha1
+kind: ScaledJob
+metadata:
+  name: azdevops-scaledjob
+spec:
+  jobTargetRef:
+    template:
+      spec:
+        containers:
+        - name: azdevops-agent-job
+          image: <azdevops-image>
+          imagePullPolicy: Always
+          env:
+          - name: AZP_URL
+            value: "<organizationUrl>"
+          - name: AZP_TOKEN
+            value: "<token>"
+          - name: AZP_POOL
+            value: "<agentpool>"
+          volumeMounts:
+          - mountPath: /var/run/docker.sock
+            name: docker-volume
+        volumes:
+        - name: docker-volume
+          hostPath:
+            path: /var/run/docker.sock
+  pollingInterval: 30
+  successfulJobsHistoryLimit: 5
+  failedJobsHistoryLimit: 5
+  maxReplicaCount: 10   
+  scalingStrategy:
+    strategy: "default"               
+  triggers:
+  - type: azure-pipelines
+    metadata:
+      poolID: "1"
+      organizationURLFromEnv: "AZP_URL"
+      personalAccessTokenFromEnv: "AZP_TOKEN"
+```
+
+### Placeholder agent
+
+You cannot queue an Azure Pipelines job on an empty agent pool because Azure Pipelines cannot validate if the pool matches the requirements for the job.
+When you try to do this you will encounter the following error:
+
+> ##[error]No agent found in pool keda-demo which satisfies the specified demands: Agent.Version -gtVersion 2.163.1
+
+You can however use a workaround to register an agent as a placeholder, you are able to queue jobs on an agent pool with no online agents.
+Make sure you don't execute any cleanup code in your container to unregister the agent when removing it to keep the placeholder agent registerd in the agent pool.
+
+### ScaledJobs in action
+
+To be able to fully create agent on-demand, a template agent was created as a placeholder to be able to queue jobs.
```

Comment:
Might be good to clarify how you have achieved this? A separate deployment or a misconfigured agent?

Reply:
No it is ok, thanks!